locust/web.py

```diff
@@ -318,17 +317,25 @@ def stats_report() -> Response:
             )
             if request.args.get("download"):
                 res = app.make_response(res)
-                res.headers["Content-Disposition"] = f"attachment;filename=report_{time()}.html"
+                host = self.environment.host or "no host"
+                res.headers["Content-Disposition"] = (
+                    f"attachment;filename=Locust - {format_safe_timestamp(self.environment.stats.start_time)} - "
+                    + f"{self.environment.locustfile} - {host}.html"
+                )
```

Comment:
I don't know under which scenario there won't be a host, but maybe in the case that there isn't a host we could exclude it?

Perhaps also consider using an underscore to separate the words as we had before? `Locust_starttime_locustfile_host.html`? Spaces aren't necessarily friendly on all systems

Reply:
I don't remember how I manage to get a null, maybe by going directly to the path without launching the scenario, but I really stumble on this case (which add a parenthesis at the end of the filename).

About the spaces vs underscore, as an old-timer I would agree, but at the current time I don't see any OS that doesn't manage spaces or utf-8 chars.

But at the end your have the final say.

Reply:
Good point about the shell: go for underscores instead of spaces.
Ok for host, but it's really an edge case that should not occur. 
I'll made a new commit soon.

Reply:
> `host = f" - {self.environment.host}" if host else ""`?

It should be `host = f"_{self.environment.host}" if self.environment.host else ""`, no ?